File: Android/melody-core/src/main/java/com/melody/core/schema/PluginManifest.kt
package com.melody.core.schema

import kotlinx.serialization.Serializable

/** Schema for a plugin's `plugin.yaml` manifest file. */
@Serializable
data class PluginManifest(
    val name: String,
    val version: String? = null,
    val description: String? = null,
    val ios: PlatformConfig? = null,
    val android: PlatformConfig? = null,
    val lua: List<String>? = null
) {
    @Serializable
    data class PlatformConfig(
        val sources: List<String>,
        val frameworks: List<String>? = null,
        val dependencies: List<String>? = null
    )
}

File: Sources/Core/Schema/Extensions/PluginManifest.swift
import Foundation

/// Schema for a plugin's `plugin.yaml` manifest file.
public struct PluginManifest: Codable, Sendable {
    public var name: String
    public var version: String?
    public var description: String?
    public var ios: PlatformConfig?
    public var android: PlatformConfig?
    public var lua: [String]?

    /// Platform-specific source files, frameworks, and dependencies for a plugin.
    public struct PlatformConfig: Codable, Sendable {
        public var sources: [String]
        public var frameworks: [String]?
        public var dependencies: [String]?
    }
}
